File: Client_Root/Client/Assets/Scripts/Network/Messages/ToRoom/EnterRoomToR.cs
using FlatBuffers;

public class EnterRoomToR : IMessage
{
    public const ushort MESSAGE_ID = MessageID.EnterRoomToR_ID;

    public string m_strPlayerKey;
    public int m_nAuthKey;
    public int m_nMatchID;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        var playerKey = m_Builder.CreateString(m_strPlayerKey);

        FBS.EnterRoomToR.StartEnterRoomToR(m_Builder);
        FBS.EnterRoomToR.AddPlayerKey(m_Builder, playerKey);
        FBS.EnterRoomToR.AddAuthKey(m_Builder, m_nAuthKey);
        FBS.EnterRoomToR.AddMatchID(m_Builder, m_nMatchID);
        var data = FBS.EnterRoomToR.EndEnterRoomToR(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.EnterRoomToR.GetRootAsEnterRoomToR(buf);

        m_strPlayerKey = data.PlayerKey;
        m_nAuthKey = data.AuthKey;
        m_nMatchID = data.MatchID;

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ToRoom/EnterRoomToR.cpp
#include "stdafx.h"
#include "EnterRoomToR.h"
#include "../../../FBSFiles/EnterRoomToR_generated.h"

EnterRoomToR::EnterRoomToR()
{
}

EnterRoomToR::~EnterRoomToR()
{
}

unsigned short EnterRoomToR::GetID()
{
	return MESSAGE_ID;
}

IMessage* EnterRoomToR::Clone()
{
	return NULL;
}

const char* EnterRoomToR::Serialize(int* pLength)
{
	auto playerKey = m_Builder.CreateString(m_strPlayerKey);

	FBS::EnterRoomToRBuilder data_builder(m_Builder);
	data_builder.add_PlayerKey(playerKey);
	data_builder.add_AuthKey(m_nAuthKey);
	data_builder.add_MatchID(m_nMatchID);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool EnterRoomToR::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::EnterRoomToR>((const void*)pChar);

	m_strPlayerKey = data->PlayerKey()->str();
	m_nAuthKey = data->AuthKey();
	m_nMatchID = data->MatchID();

	return true;
}